.github/workflows/macos-xpack-heartbeat.yml

```diff
@@ -17,15 +17,17 @@ jobs:
   macos:
     runs-on: macos-latest
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-go@v3
+    - uses: actions/checkout@v4
+    - uses: actions/setup-go@v4
       with:
         go-version-file: .go-version
     - uses: actions/setup-python@v4
       with:
         python-version: '3.9'
     - name: Install dependencies
-      run:  go install github.com/magefile/mage
+      run:  |
+        go install github.com/magefile/mage
+        cd ${{ env.BEAT_MODULE }} && npm install @elastic/synthetics
```

Comment:
Potentially, the Mage build system could be the one in charge of installing this particular dependency ... but that's something the heartbeat team might need to decide.